### PR TITLE
Upgrade graphql-tools, improve tests

### PR DESCRIPTION
**Pull request by benwarfield-usds**
## Related Issue or Background Info

- #1080 
- also the current version doesn't send an operation name with the mock graphql client for API tests, but the new one does, and that is a feature of audit logging that is otherwise difficult to test.

## Changes Proposed

- update dependency version for spring-boot-graphql-starter
- update some tests

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/api/BaseApiTest.java [hunk 1]
```diff
@@ -253,8 +245,7 @@ protected ApiAuditEvent assertLastAuditEntry(
       List<String> errorPaths) {
     ApiAuditEvent event = _auditService.getLastEvents(1).get(0);
     assertEquals(username, event.getUser().getLoginEmail());
-    // fix to come in v11.0.0 of the graphql-kickstart starter
-    // assertEquals(operationName, event.getGraphqlQueryDetails().getOperationName());
+    assertEquals(operationName, event.getGraphqlQueryDetails().getOperationName());
```
Comment (TomNUSDS) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/BaseApiTest.java:248]: nice

File: backend/src/test/java/gov/cdc/usds/simplereport/api/ApiUserManagementTest.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ void whoami_standardUser_okResponses() {
     assertUserCanAccessExactFacilities(who, Set.of(TestUserIdentities.TEST_FACILITY_2));
     assertLastAuditEntry(
         TestUserIdentities.STANDARD_USER,
-        "current-user-query-operation",
+        "WhoDat",
```
Comment (TomNUSDS) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/ApiUserManagementTest.java:58]: non-standard naming convention for graphql query action.

**Review by TomNUSDS — APPROVED** [review]
LGTM

Only minor comments. 

This is that "scary" version jump but really just seems to be synchronizing all the versions to `11` for some packages. 😟

File: backend/gradle/dependency-locks/compileClasspath.lockfile [hunk 3]
```diff
@@ -71,7 +71,6 @@ javax.xml.bind:jaxb-api:2.3.1
 net.bytebuddy:byte-buddy:1.10.18
 net.minidev:accessors-smart:1.2
 net.minidev:json-smart:2.3
-org.antlr:antlr4-runtime:4.7.2
```
[the remark this answers is not in the document]
Comment (benwarfield-usds) [reply backend/gradle/dependency-locks/compileClasspath.lockfile:74]: I suspect somebody upstream fixed their compile-time/runtime dependency configurations, and it's slowly running downhill. Not quite curious enough to check who right now. 😁

File: backend/src/test/java/gov/cdc/usds/simplereport/api/ApiUserManagementTest.java (hunk 2 above)
Comment (benwarfield-usds) [reply backend/src/test/java/gov/cdc/usds/simplereport/api/ApiUserManagementTest.java:58]: Oh, hadn't thought about that. Probably worth fixing it to not jump out as weird.

**Pull request merged**